src/TSTransformer/util/wrapReturnIfLuaTuple.ts

```diff
@@ -23,7 +24,11 @@ function shouldWrapLuaTuple(node: ts.CallExpression, exp: luau.Expression) {
 	}
 
 	// `const [a] = foo()`
-	if (ts.isVariableDeclaration(parent) && ts.isArrayBindingPattern(parent.name)) {
+	if (
+		ts.isVariableDeclaration(parent) &&
+		ts.isArrayBindingPattern(parent.name) &&
+		!arrayBindingPatternContainsHoists(state, parent.name)
+	) {
```

Comment:
This feels fragile. Is there no better way to indicate the using-ness of the LuaTuple shorthand?

Reply:
Unsure, might be worth revisiting in the future. I wasn't able to find a way right now that wasn't super hacky.